darshan-util/pydarshan/darshan/cli/summary.py

```diff
@@ -0,0 +1,235 @@
+        # the header is the application name and the log date
+        self.header = f"{app_name} ({date})"
+
+    def get_footer(self):
+        """
+        Builds the footer string for the summary report.
+        """
+        lib_ver = darshan.__version__
+        self.footer = f"Summary report generated via PyDarshan v{lib_ver}"
+
+    def get_metadata_table(self):
+        """
+        Builds the metadata table (in html form) for the summary report.
+        """
+        # assign the metadata dictionary
+        job_data = self.report.metadata["job"]
+        # TODO: once this is exposed, add the correct entry
+        log_fmt_ver = "N/A"
+        # build a dictionary with the appropriate metadata
+        metadata_dict = {
+            "Job ID": job_data["jobid"],
+            "User ID": job_data["uid"],
+            "# Processes": job_data["nprocs"],
+            "Runtime (s)": self.get_runtime(report=self.report),
+            "Start Time": datetime.datetime.fromtimestamp(job_data["start_time"]),
+            "End Time": datetime.datetime.fromtimestamp(job_data["end_time"]),
+            "Command": self.get_full_command(report=self.report),
+            "Log Filename": os.path.basename(self.log_path),
+            "Runtime Library Version": job_data["metadata"]["lib_ver"],
+            "Log Format Version": log_fmt_ver,
+        }
+        # convert the dictionary into a dataframe
+        metadata_df = pd.DataFrame.from_dict(data=metadata_dict, orient="index")
+        # write out the table in html
+        self.metadata_table = metadata_df.to_html(header=False, border=0)
+
+    def get_module_table(self):
+        """
+        Builds the module table (in html form) for the summary report.
+        """
+        # construct a dictionary containing the module names
+        # and their respective data stored in KiB
+        module_dict = {}
+        for mod in self.report.modules:
+            # retrieve the module version and buffer sizes
+            mod_version = self.report.modules[mod]["ver"]
+            # retrieve the buffer size converted to KiB
+            mod_buf_size = self.report.modules[mod]["len"] / 1024
+            # create the key/value pairs for the dictionary
+            key = f"{mod} (ver={mod_version})"
+            val = f"{mod_buf_size:.2f} KiB"
+            module_dict[key] = val
+
+        # convert the module dictionary into a dataframe
+        module_df = pd.DataFrame.from_dict(data=module_dict, orient="index")
+        # write out the table in html
+        self.module_table = module_df.to_html(header=False, border=0)
+
+    def get_stylesheet(self):
+        """
+        Retrieves the locally stored CSS.
+        """
+        # get the path to the style sheet
+        with importlib_resources.path(darshan.cli, "style.css") as path:
+            # collect the css entries
+            with open(path, "r") as f:
+                self.stylesheet = "".join(f.readlines())
+
+
+def setup_parser(parser: Any):
```

Comment:
I can't reproduce the need to use `Any` here:

```diff
--- a/darshan-util/pydarshan/darshan/cli/summary.py
+++ b/darshan-util/pydarshan/darshan/cli/summary.py
@@ -197,6 +197,7 @@ def main(args: Union[Any, None] = None):
     """
     if args is None:
         parser = argparse.ArgumentParser(description="")
+        reveal_type(parser)
         setup_parser(parser)
         args = parser.parse_args()
```

`darshan/cli/summary.py:200: note: Revealed type is "argparse.ArgumentParser"`

Reply:
I'll open an issue for a typing cleanup on this PR/code..